docs/HowToGuides/FAQ.md

```diff
@@ -0,0 +1,97 @@
+undefined symbol errors at link time.
+
+There should be a `CMakeLists.txt` in the directory where you added the new
+file, which has the list of different .h/.cpp files to be included in the build.
+Add your new file to that list.
+
+### How do I speed up iterating on changes to the build system?
+
+The general idea is to build as little as you can.
+- [Use `sccache`](/docs/HowToGuides/GettingStarted.md) or `ccache` if you
+  aren't doing so already.
+- Use `build-script`'s various `--skip-*` flags to skip configuring for
+  platforms that you do not care about.
+- If you're on macOS, use `--swift-darwin-supported-archs="x86_64"`.
+- Use a release build without assertions (`--release --no-assertions`).
+
+## Using a Locally Built Toolchain
+
+### How do I use a locally built compiler to build X?
+
+You can use the `SWIFT_EXEC` environment variable to use a locally
+built compiler to compile both packages and Xcode projects.
+
+1. For SwiftPM packages, pass the environment variable when invoking SwiftPM.
+   ```sh
+   # Assuming the current working directory contains the package, build the
+   # package using a custom compiler.
+   SWIFT_EXEC=/path/to/swiftc swift build
+   ```
+2. For Xcode projects, select the project in the Project Navigator. In the
+   Build Settings tab, click '+' and then 'Add User-Defined Setting'.
+   Create a build setting `SWIFT_EXEC` with the value set to `/path/to/swiftc`.
+   If you now do a clean build, your locally built compiler will be used.
+
+   At the time of writing, in the latest Xcode 12 beta, `SWIFT_EXEC` does not
+   work for SwiftPM integration inside Xcode, so this will not work for Xcode
+   projects that depend on SwiftPM packages.
+
+**Note:** Even thought the variable says 'SWIFT', it needs to point to
+swift**c**, not swift. The extra 'c' is not a typo.
+
+## Testing and CI
+
+### How do I reproduce/fix a test that fails in CI but passes for me locally?
+
+TODO: Write some tips here, point to Testing.md for simulator setup.
+
+## Documentation
+
+### Where can I find documentation on X?
+
+This very depends on what X is, but some broad guidelines are:
+
+1. Do a case-insensitive recursive string search.
+   - Use a specialized tool like [ripgrep](https://github.com/BurntSushi/ripgrep)
+     or [ag](https://github.com/ggreer/the_silver_searcher).
+   - Use 'Find in Workspace' in Xcode (<kbd>⌘</kbd>+<kbd>⇧</kbd>+<kbd>F</kbd>).
+   - Use `grep -i -r "mypattern" .`.
+2. Go through the [Documentation Index](/docs/README.md).
+
+## Pull Request Workflow
+
+### How do I format my changes?
+
+You should be able to install `clang-format` using your system's package
+manager. Then you can use LLVM's `git-clang-format` script to reformat your
+changes. Start out at the tip of the branch where you want to reformat the
+commits.
+
+```
+# If there is only one commit that needs to be reformatted.
+../llvm-project/clang/tools/clang-format/git-clang-format HEAD~1
```

Comment:
I'm in two minds about this.

1. I'm not sure if `clang-format` packages on all systems (say different Linux distros) also bundle `git-clang-format`.
2. If someone tries using `git help clang-format` or `git clang-format --help` those doesn't work like other git commands, you instead need to run `git-clang-format --help`, which is not necessarily obvious...

Reply:
Hrm, that's unfortunate :/ It certainly seems a shame not to mention it at all though – perhaps it could be mentioned briefly as something like the following?

```
Your package manager may also add `git-clang-format` to your path, allowing you to invoke it directly as:

git-clang-format HEAD~1
```

(Perhaps we should recommend `git-clang-format` over `git clang-format` to avoid the second issue you raise)

Reply:
I've updated this to recommend using `git-clang-format` and use the local script in case that wasn't installed.